Add StaticKit + Next.js Guide (#1157)
zeit/docs: pages/guides/deploying-statickit-with-zeit-now.mdx

  lastEdited: '2019-08-21T22:29:47.000Z'
}

[StaticKit](https://statickit.com/) is a collection of dynamic components for static sites, enabling developers to build dynamic interfaces with ease.

In this guide, you will discover how to create a simple landing page to gather email addresses using [StaticKit](https://statickit.com/) and [Next.js](https://nextjs.org/).

## Step 1: Set Up Your Next.js Project

Run the following command to create and enter into a [Next.js](https://nextjs.org/) project:

<Snippet dark text="npm create next-app next-landing-page && cd next-landing-page" />
<Caption>Bootstrapping a Next.js project with <InlineCode>create-next-app</InlineCode> and moving into the <InlineCode>/next-landing-page</InlineCode> directory.</Caption>

Replace the contents of the `/pages/index.js` file with the following code.

```jsx
import React from 'react'
import Head from 'next/head'

const OptInForm = () => {
  return (
    <form className="text-gray-800 text-lg">

>       <p className="pb-3 font-bold">Sign up to be notified when we launch.</p>

Reviewer: @msweeneydev I realized I had a typo on my example code.

Let's move the `className` stuff from the `<form>` tag above onto the `<p>` tag like this:

```
  return (
    <form>
      <p className="pb-3 font-bold text-gray-800 text-lg">
        Sign up to be notified when we launch.
      </p>
```

That way, further down when we show the diff, the `<form>` line is cleaner:

```diff
    return (
-     <form>
+     <form onSubmit={submit}>
        <p className="pb-3 font-bold text-gray-800 text-lg">
          Sign up to be notified when we launch.
        </p>
```

Author: That's clearer @derrickreimer - I've made the change thank you.

import Guide from '~/components/layout/guide'
import Snippet from '~/components/snippet'
import { InlineCode } from '~/components/text/code'
import Caption from '~/components/text/caption'
import { GenericLink } from '~/components/text/link'

export const meta = {
  title: 'Gather Emails on Your Next.js Site with StaticKit and ZEIT Now',
  description:

>     'How to use StaticKit and Next.js to collect emails from a landing page deployed with ZEIT Now.',

Reviewer 2: ```suggestion
    'Learn how to use StaticKit and Next.js to collect emails from a landing page deployed with ZEIT Now.',
```

        >
          Notify me
        </button>
      </div>
    </form>
  )
}

const Home = () => (
  <div>
    <Head>
      <title>Vaporware</title>
      <link
        href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css"
        rel="stylesheet"
      />
    </Head>
    <div className="mx-auto container px-8 py-16 sm:py-32 antialiased">
      <div className="max-w-lg mx-auto">
        <div className="flex flex-wrap items-center pb-4 text-5xl font-bold text-gray-800">
          <h1 className="mr-3">Vaporware</h1>
          <div className="mt-2 px-3 py-1 text-sm font-bold bg-orange-300 text-orange-800 rounded-full">
            Coming Soon
          </div>
        </div>
        <p className="pb-6 text-gray-700 text-lg">
          Vaporware is a fictitious app that does not yet exist. This is where
          you&rsquo;d make a compelling pitch for your new product.
        </p>
        <OptInForm />
      </div>
    </div>
  </div>
)

export default Home
```

<Caption>
  An example <InlineCode>/pages/index.js</InlineCode> file with two components.
</Caption>

> You may have noticed that the `<form>` tag is missing an `onSubmit` property, this is what you will add next.

Reviewer 2: ```suggestion
You may have noticed that the `<form>` tag is missing an `onSubmit` property, we will address that next.
```

Next, import the `useForm` hook and bind the form `onSubmit` to [StaticKit](https://statickit.com/) in your `/pages/index.js` file. Be sure to replace `[YOUR FORM ID]` with your actual form `ID` from StaticKit you received earlier.

```diff
  import React from "react";
  import Head from "next/head";
+ import { useForm } from "@statickit/react";

  const OptInForm = () => {
+   const [state, submit] = useForm("[YOUR FORM ID]");

+	  if (state.succeeded) {
+     return (
+       <p className="pb-3 font-bold text-gray-800 text-lg">
+         Thank you for signing up!
+       </p>
+     );
+   }
    return (
-     <form>
+     <form onSubmit={submit}>
        <p className="pb-3 font-bold text-gray-800 text-lg">
          Sign up to be notified when we launch.
        </p>
```

<Caption>
  Updating the <InlineCode>/pages/index.js</InlineCode> file with an onSubmit
  hook.
</Caption>

You now have a working landing page, the last step is to deploy it with [ZEIT Now](/now).

## Step 3: Deploying with ZEIT Now

If you have not yet installed Now, you can do so by [installing Now CLI](/download).

> You can now deploy your StaticKit + Next.js application library with **just a single command**:

Reviewer 2: ```suggestion
You can now deploy your StaticKit + Next.js application library with **a single command**:
```
